homoglyph.py

```diff
@@ -0,0 +1,117 @@
+import re
+import hashlib
+
+
+def str_to_hash(my_string, my_key="data", hash_size=30):
+    h = hashlib.md5(my_key.encode("utf-8"))
+    h.update(my_string.encode("utf-8"))
+    return h.hexdigest()[:hash_size]
+
+
+def homoglyph_regex(pattern):
+    """
+    Homoglyph executor for ADF Generator
+    :param str pattern: list of regex pattern in single repo adf generator specific
+    :return: list of regex pattern with replaced homoglyph patterns
+    """
+    text = pattern.get("regex", [])
+    if text:
+        for index, part_pattern in enumerate(text):
+            pattern["regex"][index] = change_dash(part_pattern)
+
+    split_dict = pattern.get("split_regex", {})
+    if split_dict and isinstance(split_dict, dict):
+        for split_keys, split_group in split_dict.items():
+            if isinstance(split_group, dict):
+                for key, value in split_group.items():
+                    pattern["split_regex"][split_keys][key] = change_dash(value)
+
+    return pattern
+
+
+def change_dash(part_pattern):
+    """
+    General function for replacing dash with list of all dash in regex pattern
+    :param str part_pattern: regex containing different types of homoglyph
+    :return: regex replaced with list of homoglyphs
+    """
+    check_alpha = r"\w-\w"
```

Comment:
replace `r"\w-\w" with \[(\w\-\w)+\]`

Reply:
@atalnarayan i was going through few regexes and I found these cases `\\d+[.;,A-Za-z]*`

Reply:
@DIVYANSH0074 do it via a checker

Reply:
@atalnarayan added the checker code for regex range and tested it o local too